x-pack/plugin/rollup/src/main/java/org/elasticsearch/xpack/downsample/MetricFieldProducer.java

```diff
@@ -54,7 +56,8 @@ void collect(Number value) {
     }
 
     @Override
-    public void collect(FormattedDocValues docValues, int docId) throws IOException {
+    public void collect(LeafFieldData leafFieldData, int docId) throws IOException {
+        final FormattedDocValues docValues = leafFieldData.getFormattedValues(fieldType.docValueFormat(null, null));
```

Comment:
I am afraid that  calling `leafFieldData.getFormattedValues()` every time we collect a field, will have a significant performance impact. Until now this was happening once per segment.